File: app/src/main/java/denis/easyweather/app/inject/modules/MainPresenterModule.java
package denis.easyweather.app.inject.modules;

import dagger.Module;
import dagger.Provides;
import denis.easyweather.app.inject.scopes.ActivityScope;
import denis.easyweather.app.main.MainPresenter;
import denis.easyweather.app.net.WeatherService;

@Module
public class MainPresenterModule {
    @Provides
    @ActivityScope
    MainPresenter provideMainPresenter(WeatherService weatherService) {
        return new MainPresenter(weatherService);
    }
}

File: app/src/main/java/denis/easyweather/app/inject/modules/MainPresenterModule.kt
package denis.easyweather.app.inject.modules

import dagger.Module
import dagger.Provides
import denis.easyweather.app.inject.scopes.ActivityScope
import denis.easyweather.app.main.MainPresenter
import denis.easyweather.app.net.WeatherService

@Module
class MainPresenterModule {
    @Provides
    @ActivityScope
    internal fun provideMainPresenter(weatherService: WeatherService): MainPresenter {
        return MainPresenter(weatherService)
    }
}
